edk2toolext/edk2_git.py

```diff
@@ -43,139 +47,170 @@ def set(self, key, value):
 
 
 class Repo(object):
```

Comment:
I find the existence of this abstraction within edk2-pytool-extensions confusing. To the casual reader, it appears to introduce redundant and narrowly scoped interfaces obscuring a more stable and broadly applicable interface directly provided by pygit2.

I understand that to some extent, it may provide minor convenience wrappers, but I don't think that justifies the additional layer of interface complexity alone.

Is it documented somewhere what value it provides over interacting with pygit2 directly?

Reply:
@makubacki This will be abandoned in favor of GitPython and completely removing the edk2_git abstraction.